app.rb

```diff
@@ -0,0 +1,220 @@
+    puts 'Name:'
+    name = gets.chomp.strip
+    puts 'Age:'
+    age = gets.chomp.to_i
+    puts 'Has parent permission? (y/n):'
+    per = gets.chomp.strip
+    per = per == 'y'
+
+    classroom = Classroom.new(clas)
+    student = Student.new(classroom, age, name, parent_permision: per)
+    @people.push(student)
+    puts "\nStudent created with ID: #{student.id}"
+    sleep 3
+    run
+  end
+
+  def create_teacher
+    puts "\n\nCreating a teacher, please enter the teacher information:"
+    puts 'Name:'
+    name = gets.chomp.strip
+    puts 'Age:'
+    age = gets.chomp.to_i
+    puts 'Specialization:'
+    specialization = gets.chomp
+
+    teacher = Teacher.new(specialization, age, name)
+    @people.push(teacher)
+    puts "\nTeacher created with ID: #{teacher.id}"
+    sleep 3
+    run
+  end
+
+  def create_person
+    puts "\n\nDo you want to create a student or a teacher?\n\nPlease, enter the number of your choice:"
+    puts '1. Student'
+    puts '2. Teacher'
+    puts "Go back to main menu: 0 \n\nEnter your option:"
+
+    option = gets.chomp.to_i
+
+    case option
+    when 1
+      create_student
+    when 2
+      create_teacher
+    when 0
+      run
+    else
+      puts "Invalid option, please try again.\n\n"
+      sleep 2
+      create_person
+    end
+  end
+
+  def create_book
+    puts "\n\nCreating a book, please enter the book information:"
+    puts 'Name:'
+    name = gets.chomp
+    puts 'Author:'
+    author = gets.chomp
+
+    book = Book.new(name, author)
+    @books.push(book)
+    puts "\nBook created successfully:\n#{name} - #{author}\n"
+    sleep 3
+    run
+  end
+
+  def select_book_from_list
+    if @books.empty?
+      puts 'There are no books yet.'
+      sleep 3
+      run
+    end
+    puts "\n\nListing all books:\n\nPlease enter the ID of the book:"
+    @books.each_with_index do |book, index|
+      puts "#{index}. #{book.title} - #{book.author}"
+    end
+    puts "\n\nEnter your option:"
+    option = gets.chomp.to_i
+    @books[option]
+  end
+
+  def select_person_from_list
+    if @people.empty?
+      puts 'There are no people yet.'
+      sleep 3
+      run
+    end
+    puts "\n\nListing all people:\n\nPlease enter the ID of the person:"
+    @people.each_with_index do |person, index|
+      puts "#{index}. #{person.id} - Nombre: #{person.name}, Edad: #{person.age}"
+    end
+    puts "\n\nEnter your option:"
+    option = gets.chomp.to_i
+    @people[option]
+  end
+
+  def create_rental
+    puts "\n\nCreating a rental, please enter the following information:"
+
+    puts 'Date:'
+    date = gets.chomp.strip
+
+    book = select_book_from_list
+    person = select_person_from_list
+
+    if book.nil? || person.nil?
+      puts "\nBook or person not found, please try again.\n\n"
+      sleep 3
+      create_rental
+    else
+      rental = Rental.new(date, book, person)
+      @rentals.push(rental)
+      puts "\nRental created successfully"
+      sleep 3
+      run
```

Comment:
You did well with your project, kindly note that according to the app implementation given, when creating a `rental`, a person should be selected by `number` and not `id` as seen in your own application. Please implement this change to follow the requirement for the project 😄 

Video screenshot :
![image](https://github.com/Ruthmy/school-library/assets/20537712/e870c994-bd87-42a7-ac9f-0d35a04079f4)

Your app :
![image](https://github.com/Ruthmy/school-library/assets/20537712/e044c590-4773-45b3-9b85-7162fa845c0f)

This is the link of the [video](https://www.youtube.com/watch?v=vkkgrhD6aXQ)

Reply:
Hi @usorfaitheloho 
Thank you for your feedback! Yes, you are right, at that point you should enter the number. In my application also works by entering the number, but by mistake I put ID in the text to display. I will fix it right away and upload the changes.
Thank you very much! ✨